src/HttpMethodOverrideListener.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * @license   http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ * @copyright Copyright (c) 2016 Zend Technologies USA Inc. (http://www.zend.com)
+ */
+
+namespace ZF\ContentNegotiation;
+
+use Zend\EventManager\AbstractListenerAggregate;
+use Zend\EventManager\EventManagerInterface;
+use Zend\Mvc\MvcEvent;
+use ZF\ApiProblem\ApiProblem;
+use ZF\ApiProblem\ApiProblemResponse;
+use Zend\Http\Request as HttpRequest;
+
+class HttpMethodOverrideListener extends AbstractListenerAggregate
+{
+    /**
+     * @var array
+     */
+    protected $httpMethodOverride = [];
+
+    /**
+     * HttpMethodOverrideListener constructor.
+     *
+     * @param array $httpMethodOverride
+     */
+    public function __construct(array $httpMethodOverride)
+    {
+        $this->httpMethodOverride = $httpMethodOverride;
+    }
+
+    /**
+     * Priority is set very high (should be executed before all other listeners that rely on the request method value).
+     * TODO: Check priority value, maybe value should be even higher??
```

Comment:
This is an appropriate value. The OPTIONS handling (i.e., HTTP method negotiation) happens at -100. Currently, we already have versioning (for Accept headers) happening at -40 (route-based version checking happens at -41). The main thing is that it happens before the OPTIONS handling, so this works fine.